src/sctools/barcode.py

```diff
@@ -0,0 +1,134 @@
+import itertools
+from collections import Counter, Mapping
+import numpy as np
+from .encodings import TwoBit
+from .stats import base4_entropy
+
+
+class BarcodeBase:
+    """Abstract base class for sets of barcodes, encoded in 2bit representation for fast operations
+
+    :method summarize_hamming_distances:  returns descriptive statistics on hamming distances
+      between pairs of barcodes
+
+    """
+
+    def __init__(self, barcode_set, barcode_length):
+        """
+        :param Counter barcode_set: dictionary
```

Comment:
Would be helpful to indicate that this is a dict of barcodes to counts. Do you need it to be a collections.Counter specifically, or any dict of barcodes to counts?

Reply:
I think any mapping to counts is probably Ok; i'll verify and make explicit.